executor/insert.go

```diff
@@ -264,6 +266,12 @@ func (e *InsertExec) batchUpdateDupRows(ctx context.Context, newRows [][]types.D
 				}
 				return err
 			}
+			// Since the temp index stores deleted key with marked 'deleteu' for unique key at the end
+			// of value, So if return a key we check and skip deleted key.
+			rowVal := val[:len(val)-1]
+			if bytes.Equal(rowVal, tables.DeleteMarkerUnique) {
+				continue
+			}
```

Comment:
We need to check this only if the key is temporary key. Otherwise, if the value of an ordinary index is "deleteux", the thing goes wrong.

Reply:
Yes only check temp index key is marked deleted, if it is ordinary index this should not be equal and it should keep on going into original check logical.

Reply:
> if it is ordinary index this should not be equal and it should keep on going into original check logical.

We had better not to make this assumption on the encoding.